openedx/features/content_tagging/rest_api/v1/tests/test_views.py

```diff
@@ -0,0 +1,964 @@
+
+        query_params = {"page": 123123}
+
+        response = self.client.get(url, query_params, format="json")
+
+        assert response.status_code == status.HTTP_404_NOT_FOUND
+
+    @ddt.data(
+        (None, "st1", status.HTTP_403_FORBIDDEN),
+        ("user", "st1", status.HTTP_200_OK),
+        ("user", "st2", status.HTTP_404_NOT_FOUND),
+        ("user", "t1", status.HTTP_200_OK),
+        ("user", "t2", status.HTTP_404_NOT_FOUND),
+        ("user", "tA1", status.HTTP_200_OK),
+        ("user", "tA2", status.HTTP_200_OK),
+        ("user", "tB1", status.HTTP_200_OK),
+        ("user", "tB2", status.HTTP_200_OK),
+        ("user", "tC1", status.HTTP_200_OK),
+        ("user", "tC2", status.HTTP_200_OK),
+        ("userA", "st2", status.HTTP_404_NOT_FOUND),
+        ("userA", "t2", status.HTTP_404_NOT_FOUND),
+        ("userS", "st2", status.HTTP_200_OK),
+        ("userS", "t2", status.HTTP_200_OK),
+    )
+    @ddt.unpack
+    def test_detail_taxonomy(self, user_attr, taxonomy_attr, expected_status):
+        taxonomy = getattr(self, taxonomy_attr)
+
+        url = TAXONOMY_ORG_DETAIL_URL.format(pk=taxonomy.id)
+
+        if user_attr:
+            user = getattr(self, user_attr)
+            self.client.force_authenticate(user=user)
+
+        response = self.client.get(url)
+        assert response.status_code == expected_status
+
+        if status.is_success(expected_status):
+            check_taxonomy_object(response.data, taxonomy)
+
+    @ddt.data(
+        (None, status.HTTP_403_FORBIDDEN),
+        ("user", status.HTTP_201_CREATED),
+        ("userA", status.HTTP_201_CREATED),
+        ("userB", status.HTTP_201_CREATED),
+        ("userC", status.HTTP_201_CREATED),
+        ("userS", status.HTTP_201_CREATED),
+    )
+    @ddt.unpack
+    def test_create_taxonomy(self, user_attr, expected_status):
+        url = TAXONOMY_ORG_LIST_URL
+
+        create_data = {
+            "name": "taxonomy_data",
+            "description": "This is a description",
+            "enabled": True,
+            "required": True,
+            "allow_multiple": True,
+        }
+
+        if user_attr:
+            user = getattr(self, user_attr)
+            self.client.force_authenticate(user=user)
+
+        response = self.client.post(url, create_data, format="json")
+        assert response.status_code == expected_status
+
+        # If we were able to create the taxonomy, check if it was created
+        if status.is_success(expected_status):
+            check_taxonomy(response.data, response.data["id"], **create_data)
+            url = TAXONOMY_ORG_DETAIL_URL.format(pk=response.data["id"])
+
+            response = self.client.get(url)
+            check_taxonomy(response.data, response.data["id"], **create_data)
+
+    @ddt.data(
+        (None, "t1", status.HTTP_403_FORBIDDEN),
+        ("user", "t1", status.HTTP_200_OK),
+        ("user", "tA1", status.HTTP_200_OK),
+        ("user", "tA2", status.HTTP_200_OK),
```

Comment:
Same here.. I think only `userS` should be allowed to update/patch/delete disabled taxonomies..

*Though I get why that's weird, since I think we let users *create* disabled taxonomies.. But we can tune this when we have a UI to support.)

Reply:
I blocked the users to update/patch/delete global taxonomies (org=None). But in this case, every user has CourseCreatorRole, so is hard to differentiate the permission from other users.

Reply:
Ah ok -- this is with the content creator groups disabled, so everyone is an editor. That makes sense.